admin/data/sql/migrations/074.do.create-auditlog-table.sql

```diff
@@ -0,0 +1,8 @@
+CREATE TABLE mtc_admin.auditLog
+(
+	id int identity primary key,
+	rowData nvarchar(max) not null,
+	tableName nvarchar(255) not null,
+	operation nvarchar(20) not null,
+	createdAt datetimeoffset default getutcdate() not null
+)
```

Comment:
How do we get user Id into this?

Reply:
that needs to be sorted before it can be classed as done.   

The most obvious approach is to ensure that all tables have a `recordedByUserId` field, and on every `INSERT` and `UPDATE` this is submitted as part of the SQL statement.  That way its captured by the trigger.  However, this doesn't provide us with a solution for `DELETE`.  Originally everything was going to be soft deletes and the mtc app user would not have any `DELETE` permissions.  This has proven to be awkward for the pupil groups and a couple of other scenarios.

It would be nice if we could somehow pass the current users id into the context of the call and capture it in the trigger.  That needs some investigation.  Any ideas welcome

Reply:
We need to pass the userId from the app to the DB, as only the app knows the user.   The app can then setup a session variable to store the user ID, which can be referenced from stored procedures.  We might need a fall back user, if one wasn't defined (like Guest) and one for the data migrations.  

It would then need to clean up the variable at the end.

Reply:
at the SQL level, the migrations run under a high privilege account, and all app users run under the same restricted db user account.  We could modify this so that each new 'app' user has a corresponding sql db user account created, which would make it very easy to capture the current user via `CURRENT_USER()`

I will research both approaches and see which turns out to be most viable

Reply:
Yes, that's the other way to do it 👍.  We would need to create a new db user whenever we saw a new user from the external auth server.   There is probably some code we add to for that.